### PR TITLE
Kanto Starter Megas

### PR DESCRIPTION
**Pull request by Qwertypop04**
## Description
Adds Mega Stones and evolutions for Venusaur, Charizard and Blastoise, using the same obtain method as the Hoenn starters. Adds Day/Night Mega Evolution for X/Y. Also slightly changes the Hoenn starter Megas to match both the Kanto starters, and the recent changes to gem shops

## Motivation and Context
We want all Megas to be available, this does so for the Kanto starter Megas in a way that matches canon and the Hoenn starters. 

## How Has This Been Tested?
Game runs, update function and battle works, stones can be bought, day/night evo for x/y works 

## Types of changes
- New content

### CODE REVIEW

**Comment by Aegyo** [comment]
Is the day/night restriction on charizard megas a canon thing?

File: src/scripts/temporaryBattle/TemporaryBattleList.ts [hunk 1]
```diff
@@ -2462,6 +2461,52 @@ TemporaryBattleList['Hoenn Stone Salesman'] = new TemporaryBattle(
...
+        new GymPokemon('Mega Charizard Y', 54478245, 63),
+        new GymPokemon('Mega Blastoise', 54478245, 63),
+    ],
+    'Wow! I\'ll let you buy some of my super fancy stones, now that I know you\'re a serious trainer!',
+    [new QuestLineStepCompletedRequirement('A Beautiful World', 12)],
+    undefined,
+    {
+        imageName: 'Owner',
+        firstTimeRewardFunction: () => {
+            if ((App.game.party.alreadyCaughtPokemonByName('Venusaur') == true) && (player.regionStarters[GameConstants.Region.kanto]() == GameConstants.Starter.Grass)) {
```
Comment (Aegyo) [inline src/scripts/temporaryBattle/TemporaryBattleList.ts:2478]: These `== true` are unnecessary

File: src/modules/items/ItemList.ts
Comment (Farboo171) [inline src/modules/items/ItemList.ts:1]: Why all the spacing changes?

**Review by Farboo171 — APPROVED** [review]
Tested:
1. Hoenn shop showing up at the right time now, and the right stones being avaliable
2. Kalos shop showing up at the right time, and the right stones becoming avaliable
3. Kalos stone reward logic for grass starter and no starter working correctly

Looks good, functionally!

File: src/modules/items/ItemList.ts
Comment (Jaaslet) [reply src/modules/items/ItemList.ts:1]: I have fixed the spacing, so all the mega stones lines up

**Review by Jaaslet — APPROVED** [review]

**Pull request merged**